uitools/toolkitcpp/cpp/Esri/ArcGISRuntime/Toolkit/FieldsPopupElementViewController.h

```diff
@@ -0,0 +1,60 @@
+/*******************************************************************************
+ *  Copyright 2012-2024 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+#ifndef ESRI_ARCGISRUNTIME_TOOLKIT_FIELDSPOPUPELEMENTVIEWCONTROLLER_H
+#define ESRI_ARCGISRUNTIME_TOOLKIT_FIELDSPOPUPELEMENTVIEWCONTROLLER_H
+
+// Toolkit headers
+#include "PopupElementViewItem.h"
+
+// Qt headers
+#include <QAbstractListModel>
+#include <QObject>
+#include <QPointer>
```

Comment:
Can this be forward declared?

Reply:
Yes, sorta, but looking at the code a little more closely it's not even needed since QPointer is included in `PopupElementViewItem.h` which is included by all PopupElementControllers because of their inheritance.

This leaves me wondering what benefit QPointer brings over just a raw pointer in this scenario. The Objects lifetime is managed by the parent which is the popup. I used it because it was used for m_popupManager in `PopupViewController.h` but it brings baggage with it and the benefits of QPointer are not being exercised. So I'm inclined to switch to just raw pointers for the stored PopupElement. Is there something I am missing here?

I don't think we can forward declare QPointer at all though since it's a template class and it requires it's full definition but I haven't worked with Template classes in a while so please correct me If I am wrong.

Reply:
You can forward declare template types.

>Is there something I am missing here?

If you aren't using the benefits of QPointer (needing to know when it's been deleted), then I'd switch to a raw pointer.